### DISCUSSION TITLE
Wedo light / sound/ potenciometer sensor

### DISCUSSION THREAD

**Discussion by Salva-OA** [pybricks/support#1970] [Ideas]
Hi,

I was looking for a wedo set as gift, and found in AliExpress some copies. In one of the sets the manufacturer/seller included 3 extra sensors: one mic, current resistor, light resistor ... since they are compatible with powerup/pf2 ports and lego original software; maybe could be included in next releases of pybricks.

If you have the capability to read what these small devices are sending to the microcontroller port, maybe we can have better control and improving build models or generate new ones; rather than emulating the originals tilt and distance, with the code you might write for their use.

Probably in the future will see also a temperature sensor as similar to was described by "o falcão", in one of his articles mixing an rcx sensor with a wedo1 tilt sensor.

Thanks for reading
Salva.

**Comment by laurensvalk** [comment]
> since they are compatible with powerup/pf2 ports and lego original software;

If this is the case, then they should work with Pybricks too without any changes. Perhaps you can use the `PUPDevice` class too.

If they don't work, we probably won't be adding support for them. It is challenging enough to support all official LEGO devices as it is.
